src/MapContainer.js

```diff
@@ -10,50 +10,78 @@
 		this.minScaleX;
 		this.maxScaleX;
 		this.maxScaleY;
-		
+
 		this.scaleX = this.minScaleX;
 		this.scaleY = this.minScaleY;
 
 		this.grid;
 
 		this.mapTiles = [];
 
-		this.addEventListeners();		
+
+		this.addEventListeners();
 	};
 
 	createjs.extend(MapContainer, createjs.Container);
 
-	MapContainer.prototype.addEventListeners = function (){
-  		this.on("mousedown", this.mouseDown, this);
-	};	
-	
-	MapContainer.prototype.loadJSON = function(){
-		if (window.File && window.FileReader && window.Blob){
+	MapContainer.prototype.TilesUpdated = function () {
+		const set = Object.keys(this.mapTiles).map(k => this.mapTiles[k].edition)
+
+
+		const newTilesets = ([...new Set(set)]).sort();
+
+		console.debug("updating", [...new Set(set)]);
+
+		const tilesetList = document.getElementById('tileset-list');
+		if (tilesetList) {
+			tilesetList.innerHTML = '';
+			const listElements = newTilesets.map(x => {
+				const li = document.createElement('li');
+				li.innerText = x;
+				return li;
+			});
+			for (const li of listElements) {
+				tilesetList.appendChild(li);
+			}
+			const legend = document.getElementsByTagName('legend')[0];
+			if (legend) {
+				legend.style.display = newTilesets.length == 0 ? 'none' : 'block'
```

Comment:
This is supposed to hide the tileset list if no tile is on the map?
If so, it doesn't work. After removing the last tile, its tileset still shows up in the legend for some reason.

Reply:
It only updates when a tile is added, not when one is removed.

Reply:
It only updates when a tile is added, not when one is removed.

Reply:
oh, thats true, maybe I can look tomorrow at that